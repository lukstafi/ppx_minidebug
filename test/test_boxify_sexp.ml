(* Test for boxify_sexp functionality - demonstrates splitting large s-expressions *)

open! Sexplib0.Sexp_conv

let () =
  Printf.printf "=== Test boxify with small threshold (boxify_sexp_from_size=2) ===\n%!";
  let _get_local_debug_runtime =
    let rt = Minidebug_db.debug_db_file ~boxify_sexp_from_size:2 "test_boxify_sexp" in
    fun () -> rt
  in
  let%debug_sexp test_boxify () : unit =
    [%log "This is the first log line"];
    [%log [ "This is the"; "2"; "log line" ]];
    [%log "This is the", 3, "or", 3.14, "log line"]
  in
  test_boxify ();
  let db = Minidebug_client.Client.open_db "test_boxify_sexp.db" in
  Minidebug_client.Client.show_trace db
    (Option.get @@ Minidebug_client.Client.get_latest_run db).run_id;
  Printf.printf "Test complete.\n%!"
