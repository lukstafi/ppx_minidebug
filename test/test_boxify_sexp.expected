=== Test structural splitting with threshold (size 10) ===
[debug] test_small @ test/test_boxify_sexp.ml:11:28-13:45
  ("This is the" 2 "log line")
[debug] test_medium @ test/test_boxify_sexp.ml:15:29-17:104
  [value] item1
    item2
    item3
    item4
    item5
    item6
    item7
    item8
    item9
    item10
[debug] test_large @ test/test_boxify_sexp.ml:19:28-21:73
  [value] <scope>
    [value] a
      1
    [value] b
      2
    [value] c
      3
    [value] d
      4
    [value] e
      5
    [value] f
      6
Test complete.
